pkg/gds/gds_test.go

```diff
@@ -105,6 +107,57 @@ func (s *gdsTestSuite) TestRegister() {
 	require.Error(err)
 }
 
+// TestVerifyContact tests that the VerifyContact RPC correctly verifies the VASP
+// against the token and sends verification emails to the admins.
+func (s *gdsTestSuite) TestVerifyContact() {
+	s.LoadFullFixtures()
+	defer s.ResetFullFixtures()
+	defer emails.PurgeMockEmails()
+	require := s.Require()
+	ctx := context.Background()
+
+	// Start the gRPC client
+	require.NoError(s.grpc.Connect())
+	defer s.grpc.Close()
+	client := api.NewTRISADirectoryClient(s.grpc.Conn)
+
+	// VASP does not exist in the database
+	request := &api.VerifyContactRequest{
+		Id:    "abc12345-41aa-11ec-9d29-acde48001122",
+		Token: "",
+	}
+	_, err := client.VerifyContact(ctx, request)
+	require.Error(err)
+
+	// Incorrect token - no verified contacts
+	request.Id = "d9da630e-41aa-11ec-9d29-acde48001122"
+	request.Token = "invalid"
+	_, err = client.VerifyContact(ctx, request)
+	require.Error(err)
+
+	// TODO: Test previously verified contact - requires modifying the fixtures to
+	// include a non-empty verification token
+
+	// Successful verification
+	request.Token = ""
+	reply, err := client.VerifyContact(ctx, request)
```

Comment:
This only works right now because we currently don't check if the input token is empty, which we probably should. We need to modify the test fixtures to include the verification token on some of the contacts so that we can properly test this.

Reply:
Should we try to address this in sc-2282 or is this a different fixture?

Reply:
Yeah we should try to address it in sc-2282.